Adding support for a new debugger (#723)
codenvy/che-docs: src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-debugger.adoc

[id="adding-support-for-a-new-debugger_{context}"]
= Adding support for a new debugger

This section describes how to add a new debugger into Eclipse Che.

[discrete]
== Prerequisites

> * Install the latest Theia plugin yeoman generator `npm install -g yo @theia/generator-plugin`

Reviewer: ```suggestion
* Install the latest Che-Theia plug-in Yeoman generator by running:
+
----
$ npm install -g yo @theia/generator-plugin
----
```

[discrete]
== Procedure

> . Generate sample by running `yo @theia/plugin`. When prompted confirm choices for “Backend plug-in”, “Samples” and then “Debugger sample”. It will create a project and a ready-to-use plugin which can be deployed.

Reviewer: ```suggestion
. Generate a sample by running `yo @theia/plugin`. When prompted, confirm the choices for *Backend plug-in*, *Samples*, and then *Debugger sample*. This creates a project and a ready-to-use plug-in, which can be deployed.
```

> . Replace the mock implementation of the debug adapter in `src/vscode-mockDebug.ts` and `src/vscode-mockRuntime.ts` by some code that talks to a "real" debugger or runtime.

Reviewer: ```suggestion
. Replace the mock implementation of the debug adapter in the `src/vscode-mockDebug.ts` and `src/vscode-mockRuntime.ts` files by code that interacts with a real debugger or runtime.
```

> . Build the plugin by running `yarn` in the root of the project.

Reviewer: ```suggestion
. Build the plug-in by running `yarn` in the root of the project.
```

> . Deploy Che Theia plugin via Che plugin-in registry.

Reviewer: ```suggestion
. Deploy the Che-Theia plug-in via the plugin-in registry.
```

[discrete]
== Additional resources

* link:https://microsoft.github.io/debug-adapter-protocol/[Debug Adapter Protocol]
* link:https://code.visualstudio.com/api/extension-guides/debugger-extension[VS Code debugger extension documentation]
* link:https://microsoft.github.io/debug-adapter-protocol/implementors/adapters/[Implementations Debug Adapters]

> * xref:adding-a-che-theia-plug-in-to-the-che-plug-in-registry[Adding a Che-Theia plug-in to the Che plug-in registry]

Reviewer: ```suggestion
* link:adding-a-che-theia-plug-in-to-the-che-plug-in-registry.html[Adding a Che-Theia plug-in to the Che plug-in registry]
```
